src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -407,6 +408,34 @@ public function fileExists(string $path, $trackContents = true): bool
         return $exists;
     }
 
+    /**
+     * Checks whether a class is available and will remain available in the "no-dev" mode of Composer.
+     *
+     * When parent packages are provided and if any of them is in dev-only mode,
+     * the class will be considered available even if it is also in dev-only mode.
+     */
+    final public function willBeAvailable(string $package, string $class, array $parentPackages): bool
+    {
+        if (!class_exists($class) && !interface_exists($class, false) && !trait_exists($class, false)) {
+            return false;
+        }
+
+        if (!class_exists(InstalledVersions::class) || !InstalledVersions::isInstalled($package) || InstalledVersions::isInstalled($package, true)) {
+            return true;
+        }
+
+        // the package is installed but in dev-mode only, check if this applies to one of the parent packages too
+
+        $rootPackage = InstalledVersions::getRootPackage()['name'] ?? '';
+        foreach ($parentPackages as $package) {
+            if ($rootPackage === $package || (InstalledVersions::isInstalled($package) && !InstalledVersions::isInstalled($package, true))) {
```

Comment:
shouldn't it be the opposite?. At least one parent package should be available in prod mode?